File: clients/kotlin/src/test/kotlin/net/eternaltwin/user/ShortUserTest.kt
package net.eternaltwin.user;

import net.eternaltwin.SerializationTestItem
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream
import kotlin.test.assertEquals

class ShortUserTest {
  companion object {
    @JvmStatic
    fun fromJsonString(): Stream<SerializationTestItem<ShortUser>> =
      SerializationTestItem.streamFromTestDir(
        "user/short-user",
        mapOf(
          "demurgos" to ShortUser(
            id = UserId("9f310484-963b-446b-af69-797feec6813f"),
            displayName = UserDisplayNameVersions(UserDisplayNameVersion(UserDisplayName("Demurgos"))),
          ),
        ),
      )
  }

  @ParameterizedTest
  @MethodSource
  fun fromJsonString(item: SerializationTestItem<ShortUser>) {
    val actual: ShortUser = ShortUser.fromJsonString(item.jsonString)
    assertEquals(item.value, actual)
  }
}

File: clients/php/test/User/ShortUserTest.php
<?php declare(strict_types=1);

namespace Eternaltwin\Test\User;

use Eternaltwin\Test\SerializationTestItem;
use Eternaltwin\User\ShortUser;
use Eternaltwin\User\UserDisplayName;
use Eternaltwin\User\UserDisplayNameVersion;
use Eternaltwin\User\UserDisplayNameVersions;
use Eternaltwin\User\UserId;
use PHPUnit\Framework\TestCase;

final class ShortUserTest extends TestCase {
  /**
   * @dataProvider provideFromJson
   * @param SerializationTestItem $item
   */
  public function testFromJson(SerializationTestItem $item): void {
    $actual = ShortUser::fromJson($item->getJson());
    $this->assertEquals($item->getValue(), $actual);
  }

  public function provideFromJson(): array {
    return SerializationTestItem::fromTestDir(
      "core/user/short-user",
      [
        "demurgos" => new ShortUser(
          UserId::fromString("9f310484-963b-446b-af69-797feec6813f"),
          new UserDisplayNameVersions(new UserDisplayNameVersion(new UserDisplayName("Demurgos")), []),
        ),
      ],
    );
  }
}
